### ISSUE TITLE
Modal component enhancements

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-react#1383]
As a consumer, I should have more flexibility over what gets displayed and what gets announced as part of a modal. 

- Title - Currently, core shows an example of a modal with no visible title, while still including a title for assistive technology in `aria-label`. A similar example should be included on the React preview page.
- Description - Currently, everything inside `.pf-c-modal-box__body` is announced, which is fine when the contents are just text, but this might not make sense if the dialog includes a lot of interactive elements. 
  - The consumer should be able to wrap a subset of contents included in `.pf-c-modal-box__body` with a `<span id="[id]">` and then pass that `[id]` value to `aria-describedby` on the `.pf-c-modal-box`.

An example of both of these use cases can be seen in [core](https://pf4.patternfly.org/components/ModalBox/examples/) in the last example labelled "ModalBox No Header Example".
